vizro-core/src/vizro/models/_action/_action.py

```diff
@@ -323,13 +387,11 @@ def build(self) -> html.Div:
             logger.debug("Callback outputs:\n%s", pformat(callback_outputs.get("external"), width=200))
 
         @callback(output=callback_outputs, inputs=callback_inputs, prevent_initial_call=True)
-        def callback_wrapper(external: Union[list[Any], dict[str, Any]], internal: dict[str, Any]) -> dict[str, Any]:
+        def action_callback(external: Union[list[Any], dict[str, Any]], internal: dict[str, Any]) -> dict[str, Any]:
             return_value = self._action_callback_function(inputs=external, outputs=callback_outputs.get("external"))
             if "external" in callback_outputs:
-                return {"internal": {"action_finished": None}, "external": return_value}
-            return {"internal": {"action_finished": None}}
-
-        return html.Div(id=f"{self.id}_action_model_components_div", children=self._dash_components, hidden=True)
+                return {"internal": {"action_finished": time.time()}, "external": return_value}
+            return {"internal": {"action_finished": time.time()}}
```

Comment:
N.B. the `_define_callback` (was `build`) method no longer returns `self._dash_components`. These are obtained directly in `Page.build`.